cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/internal/AliasedSSA.qll

```diff
@@ -81,8 +81,11 @@ private newtype TMemoryLocation =
   TAllNonLocalMemory(IRFunction irFunc, boolean isMayAccess) {
     isMayAccess = false or isMayAccess = true
   } or
-  TAllAliasedMemory(IRFunction irFunc, boolean isMayAccess) {
-    isMayAccess = false or isMayAccess = true
+  TAllAliasedMemory(IRFunction irFunc, boolean isMayAccess, boolean canDefineReadOnly) {
+    isMayAccess = false and
+    (canDefineReadOnly = true or canDefineReadOnly = false)
```

Comment:
```suggestion
    canDefineReadOnly = [true, false]
```
I think this is easier to read, making it clear that it's the same variable in both cases. That's in contrast to the line below, where it's different variables bound to true and false.